Command to migrate MariaDB database

amyfromandi opened the pull request:
This branch contains migration code that enables a complete MariaDB to PostgreSQL database migration, as detailed in #60. The script in this pull request should make several key data evolutions:

- [x] Migrate the `macrostrat` MariaDB database to PostgreSQL, retaining all database tables, rows, etc.
- [x] Merge the resulting database (`macrostrat_two`, currently) into Macrostrat's PostgreSQL database/`macrostrat` schema, overwriting tables if needed
- [x] https://github.com/UW-Macrostrat/macrostrat/pull/73: Run PostgreSQL migrations to ensure all constraints/sequences/foreign keys are defined
- [x] https://github.com/UW-Macrostrat/macrostrat/issues/80: ~~Regenerate foreign keys from other schemas (e.g., `maps`) as needed.~~ This has been moved to future work, based on some complexities identified by @mwestphall.

A successful operation of this script will result in full integration of all data from MariaDB, allowing the Macrostrat API to operate off of a single PostgreSQL database (UW-Macrostrat/macrostrat-api#229)

Architecturally, this should be implemented within the `macrostrat` command-line application:

- [x] https://github.com/UW-Macrostrat/macrostrat/pull/74:
    - [x] Implement as a `macrostrat` subcommand, e.g., `macrostrat migrate-mariadb` (we can figure out the "right" name later)
    - [x] Use database credentials, etc. from `macrostrat.toml` rather than replicating them elsewhere
    - [x] Use the `macrostrat.database` module for running SQL commands
    - [x] As much as possible, we should use Docker for command-line processing
- [x] Once this is all validated, we can remove the `macrostrat v1 schlep` scripts that were the initial version of this process

### Latest update:
* restored maria-migrate branch
* added .gitignore file
* refactored `compare_data_counts()` to compare any two databases passed through as parameters.

### Next steps:
* Resolve all discrepancies between the PostgreSQL **macrostrat_two** database and the development/production **macrostrat** PostgreSQL database.

davenquinn commented:
For data validation summary with these scripts (see #60), can we print row counts, etc. as tables to improve readability (maybe using the `rich` Python library)?

amyfromandi commented:
Removed v1 schlep scripts and moved all index files here [schlep-index](https://github.com/UW-Macrostrat/macrostrat/blob/maria-migrate/cli/macrostrat/cli/database/mariadb/postgresql_migration/schlep-index.sql).
Kept schlep scripts for tables that are modified in the pre/post migration scripts in case they are needed for future references i.e. the process scripts. [remaining schlep scripts](https://github.com/UW-Macrostrat/macrostrat/tree/maria-migrate/cli/macrostrat/cli/commands/table_meta)
